p2p/README.md

```diff
@@ -67,105 +66,431 @@ flowchart TD
+    }
+
+  class Bus {
+    <<interface>>
+    PublishEventToBus(PocketEnvelope)
+    GetBusEvent() PocketEnvelope
+  }
+  p2pModule --> Bus
+
+    class BackgroundRouter {
+        UnicastRouter
+        -handler MessageHandler
+        +Broadcast([]byte) error
+        -handleBackgroundMsg([]byte) error
+        -readSubscription(subscription *pubsub.Subscription)
+    }
+
+    class UnicastRouter {
+        -messageHandler MessageHandler
+        -peerHandler PeerHandler
+        +Send([]byte, address Address) error
+        -handleStream(stream libp2pNetwork.Stream)
+        -readStream(stream libp2pNetwork.Stream)
+    }
+    RainTreeRouter --* UnicastRouter : (embedded)
+    BackgroundRouter --* UnicastRouter : (embedded)
+
+    p2pModule ..* RainTreeRouter
+    RainTreeRouter --o RainTreeMessage
+    
+    p2pModule ..* BackgroundRouter
+    BackgroundRouter --o BackgroundMessage
+
+    p2pModule --o PocketEnvelope
+    p2pModule --* NonceDeduper
+```
 
-hs --4a--> rRT
-hs --4b--> rBG
-rBG  --"5a (cont. propagation)"--> rHost
-linkStyle 11 stroke:#ff3
-rRT  --"5b (cont. propagation)"--> rHost
-linkStyle 12 stroke:#ff3
-end
+### Peer Discovery
+Peer discovery involves pairing peer IDs to their network addresses (multiaddr).
+This pairing always has an associated TTL (time-to-live), near the end of which it must
+be refreshed.
+
+In the background gossip overlay network (`backgroundRouter`), peers will re-advertise themselves 7/8th through their TTL.
+This refreshes the libp2p peerstore automatically.
+
+In the raintree gossip overlay network (`raintreeRouter`), the libp2p peerstore is **NOT** currently refreshed _(TODO: [#859](https://github.com/pokt-network/network/isues/859))_.
+
+```mermaid
+flowchart TD
+  subgraph bus
+  end
+  
+  subgraph pers[Persistence Module]
+  end
+
+  subgraph cons[Consensus Module]
+  end
 
-lHost --2--> rHost
+  cons -- "(staked actor set changed)\npublish event" --> bus
+  bus --> rPM
+  rPM -- "get staked actors\nat current height" --> pers
+
+  subgraph p2p["P2P Module"]
+    host[Libp2p Host]
+    host -- "incoming\nraintree message" --> rtu
+    host -- "incoming\nbackground message" --> bgu
+    host -- "incoming\ntopic message" --> bgr
+    host -- "DHT peer discovery" --> rDHT
+  
+    subgraph rt[RainTree Router]
+      subgraph rPS[Raintree Peerstore]
+        rStakedPS([staked actors only])
+      end
+
+      subgraph rPM[PeerManager]
+      end
+
+      rtu[UnicastRouter]
+      
+      rPM -- "synchronize\n(add/remove)" --> rPS
+      rtu -. "(no discovery)" .-x rPS
+    end
+
+    subgraph bg[Background Router]
+      subgraph rBGPS[Background Peerstore]
+        rNetPS([all P2P participants])
+      end
+
+      subgraph bgr[GossipSub Topic\nSubscription]
+      end
+
+      subgraph rDHT[Kademlia DHT]
+      end
+
+      bgu -- "add if new" --> rBGPS
+      bgr -- "add if new" --> rBGPS
+      rDHT -- "continuous import" --> rBGPS
+         
+      bgu[UnicastRouter]
+    end
+
+  end
 ```
```

Comment:
I glossed over the remaining diagrams and makes sense to me (at a high level) but just notifying that there might be micro-typos I haven't caught.

It'll be super useful for anyone (myself included) when debugging and jumping into P2P